crates/vm/src/wasm/compilation_cache/common.rs

```diff
@@ -53,8 +54,15 @@ pub trait CacheName: Clone + std::fmt::Debug {
     fn name() -> &'static str;
 }
 
+/// WASM cache key consists of the code hash and gas meter kind.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct CacheKey {
+    code_hash: Hash,
+    gas_meter_kind: GasMeterKind,
```

Comment:
Aren't wasm modules expected to be always compiled with `GasMeterKind::MutGlobal`? Or do we use the native meter at times? It looks like even for governance proposals we use the global variable

Reply:
the only use so far is in the eval vp host fn, I think. originally I discussed with tomas using it for dry runs, as well, but I think ultimately we decided against it. the reasoning behind keeping the host fn gas meter was to allow running arbitrary untrusted wasms, which could mess with the gas global value, to achieve infinite execution times